### PR TITLE
Add support for running Action on GitHub `push` event :christmas_tree: 

### PR DESCRIPTION
**Pull request by jamacku**
- Limitations: It doesn't work for `force-push`
- #166

### TODO:

- [x] CHANGELOG
- [x] Documentation
- [x] Tests
- [x] Manual Testing
- [x] Implementation

### CODE REVIEW

File: .github/workflows/differential-shellcheck.yml [hunk 1]
```diff
@@ -18,6 +18,12 @@ jobs:
       pull-requests: write
 
     steps:
+      # Let's have this log of GitHub context for further development and debugging
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
```
Comment (lzaoral) [inline .github/workflows/differential-shellcheck.yml:25]:
```suggestion
        run: echo "${{ toJson(github) }}"
```
Wouldn't something like this work as well?

Comment (jamacku) [reply .github/workflows/differential-shellcheck.yml]: Good question, it sort of works, but it produces [an error](https://github.com/redhat-plumbers-in-action/differential-shellcheck/actions/runs/3738787528/jobs/6345243889#step:2:707). I prefer to stick to the current approach, even tho it's a bit longer.

```sh
/home/runner/work/_temp/07bef2ce-3101-493f-ac3d-bfd545b66ede.sh: line 20: push: command not found
```

Comment (lzaoral) [reply .github/workflows/differential-shellcheck.yml:25]: My bad, try enclosing the command in single quotes, e.g. `'text'`. I forgot that JSON uses double quotes as a key delimiters.

**Comment by jamacku** [comment]
All reported ShellCheck defects are inside tests which are OK for now. In follow-up PR, we will introduce full scans, which are required for `differential-shellcheck` to work correctly on the `push` event.

**Pull request merged**